### DISCUSSION TITLE
Support for Eigen::Vector

### DISCUSSION THREAD

**Discussion by solosuper** [alandefreitas/matplotplusplus#140] [Ideas] [answered]
Hi,
Have you considered support for Eigen data types as input to the `plot` command?

**Comment by alandefreitas** [comment, marked as answer]
Hi @solosuper ,

The plots are supposed to work for any range (something with `begin()` and `end()`) or range or ranges of object that are ultimately convertible to `double`: 

https://github.com/alandefreitas/matplotplusplus/blob/87ff5728b14ad904bc6acaa2bf010c1a03c6cb4a/source/matplot/core/axes_type.h#L1725

**Reply by solosuper** [reply]
Ah, that's great!
If anyone else sees this, [Eigen supports this from version 3.4](https://eigen.tuxfamily.org/dox-devel/group__TutorialSTL.html) :+1:
